pkg/controller/daemon/daemoncontroller.go

```diff
@@ -763,6 +784,12 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 	if err != nil {
 		return fmt.Errorf("unable to retrieve ds %v from store: %v", key, err)
 	}
+	// DeepCopy so we don't mutate objects in the cache.
+	cp, err := api.Scheme.DeepCopy(ds)
```

Comment:
This is likely a large performance change.  It's correct, but it's also really late to be significantly altering the controller impact like this.